packages/core/src/app/serialization/serializers.ts

```diff
@@ -22,6 +23,30 @@ export const deserializeObject = (obj: string): {} => {
   return JSON.parse(obj)
 }
 
+/**
+ * Serializes the provided object to its canonical hex string representation.
+ *
+ * @param obj The object to serialize.
+ * @returns The serialized object as a string.
+ */
+export const serializeObjectAsHexString = (obj: {}): string => {
+  const stringified: string = JSON.stringify(obj)
+  const asHexString = '0x' + Buffer.from(stringified, 'utf-8').toString('hex')
+  return asHexString
+}
+
+/**
+ * Deserializes the provided hex string into its object representation.
+ * This assumes the string was serialized using the associated serializer.
+ *
+ * @param obj The string to deserialize.
+ * @returns The deserialized object.
+ */
+export const deserializeObjectAsHexString = (obj: string): {} => {
+  const asBuffer = hexStrToBuf(obj).slice(2)
```

Comment:
use `remove0x` instead so that in the event that it doesn't start with `0x`, we don't remove the first byte.